### DISCUSSION TITLE
Micropython 1.22 - ESP32 PSRAM support

### DISCUSSION THREAD

**Discussion by Pawelinode** [micropython/micropython#12934] [ESP32]
In older versions of Micropython for ESP32, there were projects "GENERIC_SPIRAM" and "GENERIC". In version 1.22 there is only "GENERIC".
After entering this version into ESP32-PICO-MINI-02, additional RAM is not detected. I have enabled SPI RAM support in the SDK config. During compilation I get the error `iram0_0_seg' exceeded by 11848 bytes. How to get a Micropython image with PSRAM support ?

**Comment by robert-hh** [comment]
Besides the basic version, there are versions for boards with psram. Just scroll down at the page for GENERIC boards.

**Reply by Pawelinode** [reply]
Is it about this folder https://github.com/micropython/micropython/tree/master/ports/esp32/boards ?

**Reply by robert-hh** [reply]
In e.g. this GENERIC folder, you have  a file board.json, which lists the variants of for this build. You will find the pre-built variants e.g. here: https://micropython.org/download/ESP32_GENERIC/
If you want to build a variant yourself, call e.gf. `make BOARD_VARIANT=SPIRAM BOARD=GENERIC`.

**Reply by Pawelinode** [reply]
Thank you very much for your help. Now everything works fine.
